tests/units/units.cpp

```diff
@@ -2469,3 +2469,52 @@ TEST(Units, addUnitsMultipleTimes)
     // units.
     EXPECT_EQ(size_t(1), model->unitsCount());
 }
+
+TEST(Units, childUnitsWithIllegalPrefix)
+{
+    auto model = libcellml::Model::create("model");
+
+    auto u1 = libcellml::Units::create("u1");
+    auto u2 = libcellml::Units::create("u2");
+    auto u3 = libcellml::Units::create("u3");
+    auto u4 = libcellml::Units::create("u4");
+    auto u5 = libcellml::Units::create("u5");
+
+    model->addUnits(u1);
+    model->addUnits(u2);
+    model->addUnits(u3);
+    model->addUnits(u4);
+    model->addUnits(u5);
+
+    u5->addUnit("u4");
+    u4->addUnit("u3");
+    u3->addUnit("u2", "notAPrefix");
+    u2->addUnit("second");
+    u1->addUnit("second", "milli");
+
+    auto scaling = libcellml::Units::scalingFactor(u1, u5);
+    EXPECT_EQ(0.0, scaling);
+}
+
+TEST(Units, scalingFactorBetweenSameUnits)
+{
+    auto model = libcellml::Model::create("model");
```

Comment:
What is model doing here?

Reply:
See line 196 in src/units.cpp.  In order to check for child unit references, we need to have access to the other units in the model... so it needs a model.

Reply:
But the model is not used by anything?

Reply:
Duh! I get it now.  Changed in a8c89fdb